service/geopmdpy/access.py

```diff
@@ -207,13 +250,46 @@ def get_group_controls(self, group):
         _, all_controls = self._geopm_proxy.PlatformGetGroupAccess(group)
         return '\n'.join(all_controls)
 
-    def read_stdin(self):
-        """Parse list of signals or controls from standard input
+    def edited_names_path(self, initial_names):
+        default_editor = '/usr/bin/vi'
+        if not os.path.exists(default_editor):
+            default_editor = '/bin/vi'
+        editor = os.environ.get('EDITOR', default_editor)
+        fid, path = tempfile.mkstemp('geopm-service-access-tmp')
+        os.close(fid)
+        with open(path, 'w') as fid:
+            fid.write(initial_names)
+        try:
+            subprocess.run([editor, path], check=True)
+        except subprocess.CalledProcessError:
+            os.unlink(path)
+            raise
+        return path
+
+    def edited_names(self, is_control, group):
+        if is_control:
+            initial_names = self.get_group_controls(group)
+        else:
+            initial_names = self.get_group_signals(group)
+        edit_path = self.edited_names_path(initial_names)
+        try:
+            with open(edit_path) as fid:
+                names = self.read_names(fid)
+        finally:
+            os.unlink(edit_path)
```

Comment:
Could potentially avoid manual `os.unlink` and `os.close` calls here and in `edited_names_path` if this block is instead wrapped in a `with tempfile.NamedTemporaryFile()` context and the tempfile.name path is passed as an argument to `edited_names_path`.

Reply:
I had some trouble implementing that without having the file open in python while the editor also had it open.  Because the unlink() is bound to the close() call when you use the ``__enter__()``/``__exit__()`` interfaces through ``with`` you always get both calls.  In this case we would like to close() the file in python without unlink() while the editor process modifies the file.  After the file is open() again in python we will unlink() it when we close() the second time.

Reply:
Okay, no need to close the file while it is edited.  Thanks for the suggestion.